invenio_app_ils/documents/serializers/document.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2018 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Document serializers."""
+
+from flask import current_app
+from invenio_records_rest.serializers.json import JSONSerializer
+
+
+class DocumentJSONSerializer(JSONSerializer):
+    """Serialize document."""
+
+    def transform_record(self, pid, record, links_factory=None, **kwargs):
+        """Transform record into an intermediate representation."""
+        document = super().transform_record(
+            pid,
+            record,
+            links_factory=links_factory,
+            **kwargs
+        )
+        document = self.build_cover_urls(document)
+        return document
+
+    def transform_search_hit(self, pid, record_hit, links_factory=None,
+                             **kwargs):
+        """Transform search result hit into an intermediate representation."""
+        hit = super().transform_search_hit(
+            pid,
+            record_hit,
+            links_factory=links_factory,
+            **kwargs
+        )
+        hit = self.build_cover_urls(hit)
+        return hit
+
+    def build_cover_urls(self, document):
+        """Build cover urls for document."""
+        url_builder = current_app.config.get("ILS_BUILD_LITERATURE_COVER_URLS")
```

Comment:
minor: IMHO, I like the `builder` name that you used!
```suggestion
        url_builder = current_app.config.get("ILS_LITERATURE_COVER_URLS_BUILDER")
```